src/usr/hdat/hdatiplparms.C

```diff
@@ -724,18 +725,97 @@ void HdatIplParms::hdatGetSystemParamters()
 
     this->iv_hdatIPLParams->iv_sysParms.hdatSplitCoreMode = 1;
 
-    TARGETING::ATTR_SYSTEM_BRAND_NAME_type l_systemBrandName = {0};
-    if(l_pSysTarget->tryGetAttr<TARGETING::ATTR_SYSTEM_BRAND_NAME>
-                                                         (l_systemBrandName))
+    /* Read UTIL::F5 keyword (maxlen=16)*/
+    errlHndl_t l_err = NULL;
+    size_t  dataSize = 0;
+    char    dataBuffer[17]={0};
+    char    sysVendorname[64]={0};
+    Target* node_tgt = UTIL::getCurrentNodeTarget();
+
+    l_err = deviceRead(node_tgt, NULL, dataSize, DEVICE_PVPD_ADDRESS( PVPD::UTIL, PVPD::F5 ));
+
+    if (l_err || (dataSize > sizeof(dataBuffer) - 1))
     {
-        strcpy(reinterpret_cast<char*>
-                   (this->iv_hdatIPLParams->iv_sysParms.hdatSystemVendorName),
-                                                           l_systemBrandName);
+        HDAT_ERR("Read UTIL::F5 length error, len=%d...", dataSize);
+    }
+    else
+    {
+        memset(dataBuffer, 0, sizeof(dataBuffer));    
+        l_err = deviceRead(node_tgt, dataBuffer, dataSize, DEVICE_PVPD_ADDRESS( PVPD::UTIL, PVPD::F5 ));
+        if (l_err)
+        {
+            HDAT_ERR("Read UTIL::F5 data error");
+        }
+        else if(strlen(dataBuffer) > sizeof(sysVendorname) - 1)
+        {
+            HDAT_ERR("Read UTIL::F5 string len override size: %d", strlen(dataBuffer));
+        }
+        else
+        {
+            HDAT_DBG("Read UTIL::F5 success len=%d str=%s", dataSize, dataBuffer);
+            if (strlen(dataBuffer))
+            {
+                strcat(sysVendorname, dataBuffer);
+            }
+        }
+    }
+
+    /* Read UTIL::F6 keyword(maxlen=16) */
+    dataSize = 0;
+    l_err = deviceRead(node_tgt, NULL, dataSize, DEVICE_PVPD_ADDRESS( PVPD::UTIL, PVPD::F6 ));
```

Comment:
All same comments from lines from 735 through 761 apply to lines 765 through 795

Reply:
done